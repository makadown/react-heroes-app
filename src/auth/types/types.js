/**
 * Contiene nombres de las acciones que podr√© "disparar"
 */
export const types = {
    login: '[Auth] Login',
    logout: '[Auth] Logout',
}