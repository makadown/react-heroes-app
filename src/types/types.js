export const Types = {
    login: '[auth] login',
    logout: '[auth] logout',
};