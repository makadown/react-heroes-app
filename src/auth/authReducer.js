import { Types } from '../types/types';

export const AuthReducer = (state = {}, action) => {
    switch (action.type) {
        case Types.login:
            return {
                ...action.payload,
                logged: true
            };
        case Types.logout:
            return {
                logged: false
            };

        default:
            return state;
    }
};