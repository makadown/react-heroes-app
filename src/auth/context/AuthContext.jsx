import React, { createContext } from 'react'

/**
 * Contexto de autenticaci√≥n usado por AuthProvider.jsx
 */
export const AuthContext = createContext();
