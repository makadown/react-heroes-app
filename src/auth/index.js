export { LoginPage } from './pages/LoginPage';
export {AuthContext} from './context/AuthContext';
export {AuthProvider} from './context/AuthProvider';
export {authReducer} from './context/authReducer';
export {types} from './types/types';