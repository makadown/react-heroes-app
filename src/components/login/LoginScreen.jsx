import React from 'react';
import { useForm } from 'react-hook-form';

export const LoginScreen = ({ history }) => {
  const { register, handleSubmit } = useForm();
  const onSubmit = (data) => {
    //  alert(JSON.stringify(data));
    if ( !(data.username === 'test' && data.password === 'test')) {
      alert(
        'Credenciales incorrectas. Capture "test"/"test" para este demo =)'
      );
    } else {
      history.replace('/');
    }
  };

  return (
    <>
      <div className="sidenav">
        <div className="login-main-text">
          <h2> Heroes App </h2>
          <p>Introduce tus credenciales.</p>
        </div>
      </div>
      <div className="main">
        <div className="col-md-6 col-sm-12">
          <div className="login-form">
            <form onSubmit={handleSubmit(onSubmit)}>
              <div className="form-group">
                <label>User Name</label>
                <input
                  type="text"
                  {...register('username')}
                  className="form-control"
                  placeholder="User Name"
                />
              </div>
              <div className="form-group">
                <label>Password</label>
                <input
                  type="password"
                  {...register('password')}
                  className="form-control"
                  placeholder="Password"
                />
              </div>
              <button type="submit" className="btn btn-secondary">
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </>
  );
};
