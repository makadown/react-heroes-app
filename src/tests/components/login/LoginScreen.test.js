import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter, Route, Router } from 'react-router';
import { AuthContext } from '../../../auth/AuthContext';
import { useForm } from 'react-hook-form';
import { Types } from '../../../types/types';
import { LoginScreen } from '../../../components/login/LoginScreen';

describe('Pruebas de <LoginScreen/>', () => {

    // const { register, handleSubmit } = useForm();

    const historyMock = {
        length: 10,
        goBack: jest.fn(),
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn(),
        replace: jest.fn()
    };

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false
        }
    };

    /* const data = {
         username: 'test',
         password: 'test'
     }*/

    const useForm = {
        dispatch: jest.fn(),
        handleSubmit: jest.fn(),
        user: {
            logged: false
        }
    };

    test('debe de verse', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <LoginScreen history={historyMock}></LoginScreen>
                </MemoryRouter>
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });

    // Este no jala porque NO hice el jsx justo como en el curso jijiji
/*
    test('debe de realizar dispatch y navegacion', () => {

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
                <MemoryRouter>
                    <LoginScreen history={historyMock}></LoginScreen>
                </MemoryRouter>
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        wrapper.find('button').simulate('click', {
            data: {
                username: 'test',
                password: 'test'
            }
        });
        expect(useForm.handleSubmit).toHaveBeenCalled();
    });*/

});
