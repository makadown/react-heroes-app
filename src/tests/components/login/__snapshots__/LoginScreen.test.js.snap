// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <LoginScreen/> debe de realizar dispatch y navegacion 1`] = `
Array [
  <div
    className="sidenav animate__animated animate__fadeInLeft"
  >
    <div
      className="bg"
    />
    <div
      className="login-main-text"
    >
      <h2>
         Heroes App 
      </h2>
      <p>
        Introduce your credentials.
      </p>
    </div>
  </div>,
  <div
    className="main animate__animated animate__fadeInRight"
  >
    <div
      className="col-md-6 col-sm-12"
    >
      <div
        className="login-form"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label>
              User Name
            </label>
            <input
              className="form-control"
              name="username"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="User Name"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="btn btn-secondary"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;

exports[`Pruebas de <LoginScreen/> debe de verse 1`] = `
Array [
  <div
    className="sidenav animate__animated animate__fadeInLeft"
  >
    <div
      className="bg"
    />
    <div
      className="login-main-text"
    >
      <h2>
         Heroes App 
      </h2>
      <p>
        Introduce your credentials.
      </p>
    </div>
  </div>,
  <div
    className="main animate__animated animate__fadeInRight"
  >
    <div
      className="col-md-6 col-sm-12"
    >
      <div
        className="login-form"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label>
              User Name
            </label>
            <input
              className="form-control"
              name="username"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="User Name"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label>
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
          <button
            className="btn btn-secondary"
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
